### ISSUE TITLE
Panics on Bare Repo

### ISSUE THREAD

**Issue by kylemacey** [nickgerace/gfold#11]
Hey! I came to your tool from [📝  How I Read "The Rust Programming Language"](https://nickgerace.dev/post/how-i-read-the-rust-programming-language). 

I noticed that the binary panics on bare repositories.

![image](https://user-images.githubusercontent.com/519171/90809470-85ea3000-e2ef-11ea-98fc-ed21a6bc5441.png)

<details>

```
➜  tmp RUST_BACKTRACE=full /Users/kyle/.cargo/bin/gfold
thread 'main' panicked at 'failed get statuses: cannot status. This operation is not allowed against bare repositories.; class=Repository (6); code=BareRepo (-8)', src/gfold.rs:71:27
stack backtrace:
   0:        0x1067e61fe - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h13e6f26430148ff2
   1:        0x1068038cc - core::fmt::write::h45a9bd04db15c24f
   2:        0x1067e3629 - std::io::Write::write_fmt::h8ae61696f13218d5
   3:        0x1067e7e75 - std::panicking::default_hook::{{closure}}::hc98fe2390b6284e5
   4:        0x1067e7bb2 - std::panicking::default_hook::h9a63cea866a4c14a
   5:        0x1067e83d5 - std::panicking::rust_panic_with_hook::h4d446ca45c8e1faa
   6:        0x1067e7fa2 - rust_begin_unwind
   7:        0x10680f1db - std::panicking::begin_panic_fmt::h0f7b83f993394bc5
   8:        0x1066fedb0 - gfold::gfold::walk_dir::h7ce7a797048a9dd9
   9:        0x1066fc1b9 - gfold::main::h9a34aaa0573dc180
  10:        0x1066fc4e6 - std::rt::lang_start::{{closure}}::h1185fef6ccb1df63
  11:        0x1067e86a9 - std::rt::lang_start_internal::hecbcda5e1a75a865
  12:        0x1066fc4d9 - main
```

</details>

Steps to reproduce 

```shell
$ mkdir repos
$ cd repos
$ git clone --bare https://github.com/octocat/Spoon-Knife
$ gfold
```

**Comment by nickgerace** [comment]
Thanks for filing the issue @kylemacey. This is definitely a bug. I sincerely appreciate the detailed breakdown here; really informative, and your breakdown helped me figure out where it occurs.

I'll fix this bug when I get the chance, and will update the issue down the road. Stay tuned!

**Comment by nickgerace** [comment]
I'd added a fix for now. Part of me wants to pursue adding a "bare" label to the final table, but I'd like to at least put in this fix first.

**Comment by nickgerace** [comment]
Got some awesome mentorship from @yaahc. Made an enhancement to the original fix with PR #15. Leaving this closed! Thank you again @kylemacey.
